numpy/core/src/multiarray/datetime.c

```diff
@@ -2655,8 +2659,10 @@ convert_pyobject_to_timedelta(PyArray_DatetimeMetaData *meta, PyObject *obj,
             return 0;
         }
     }
+
     /* Do no conversion on raw integers */
-    else if (PyInt_Check(obj) || PyLong_Check(obj)) {
+    npy_intp intp = PyArray_PyIntAsIntp(obj);
```

Comment:
I've just realized that this isn't going to work on platforms with `intp == int32`...

Reply:
Do you mean that it doesn't work on 32bit platforms?

Reply:
I mean that if `sizeof(npy_intp) < sizeof(npy_int64)`, then this will fail for large times.

Reply:
Can you give an example, please? Cause I can't reproduce this bug (I get sizeof(npy_intp) == 8 on all tests)

Reply:
Then on your platform, everything is ok. There exist platforms where `sizeof(npy_intp) == 4`, and on those platforms things will fail.